### PR TITLE
Issue 229

### PR DESCRIPTION
**Pull request by boehsermoe**
Unit test for exception handling while proxy data sync. #229

### CODE REVIEW

File: src/proxy/ClientTable.php [hunk 1]
```diff
@@ -3,8 +3,10 @@
 namespace luya\admin\proxy;
 
 use Curl\Curl;
+use Symfony\Component\Debug\Exception\UndefinedMethodException;
```
Comment (nadar) [inline src/proxy/ClientTable.php]: @boehsermoe This seems to be not used in the context.

File: src/proxy/ClientTable.php [hunk 2]
```diff
@@ -3,8 +3,10 @@
...
 use Yii;
 use yii\base\BaseObject;
+use yii\db\Exception;
```
Comment (nadar) [inline src/proxy/ClientTable.php]: At first sight from the commit diff, this also seems not to be used in the context.

File: tests/admin/proxy/ClientTableTest.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        $this->app->db->createCommand('CREATE TABLE IF NOT EXISTS temp_synctest LIKE admin_user')->execute();
+        
+        $ctrl = new ProxyController('proxyctrl', $this->app);
+        $build = new ClientBuild($ctrl, [
+            'buildConfig' => ['tables' => ['temp_synctest' => ['name' => 'temp_synctest']]],
+        ]);
+        $table = new ClientTableMock($build, ['name' => 'temp_synctest']);
+    
+        try {
+            $table->syncData();
```
Comment (nadar) [inline tests/admin/proxy/ClientTableTest.php]: you don't need try catch block as far as i know, you can just add 

$this->expectException('yii\db\Exception');

before the method which will thrown the exception.

https://phpunit.de/manual/6.5/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.exceptions

Comment (nadar) [reply tests/admin/proxy/ClientTableTest.php]: This should be done thought, then we can merge

**Comment by nadar** [comment]
The unit tests seems to have another exception as expected:

> Failed asserting that exception of type "yii\base\ErrorException" matches expected exception "yii\db\Exception". Message was: "Undefined index: rows" at

**Comment by nadar** [comment]
It seems there is still a problem regarding this PR:

Failed asserting that exception of type "yii\base\ErrorException" matches expected exception "yii\db\Exception". Message was: "Array to string conversion" at
/home/travis/build/luyadev/luya-module-admin/src/proxy/ClientTable.php:126

(the other failures are a problem from the master)

**Comment by boehsermoe** [comment]
I think we finally have it 😓

**Comment by nadar** [comment]
thanks!

**Pull request merged**